credsweeper/credentials/line_data.py

```diff
@@ -87,14 +87,22 @@ def get_span_from_match_obj(_match_obj: re.Match, group: str) -> Tuple[int, int]
         self.key = get_group_from_match_obj(match_obj, "keyword")
         self.separator = get_group_from_match_obj(match_obj, "separator")
         self.separator_start, self.separator_end = get_span_from_match_obj(match_obj, "separator")
-        self.value = get_group_from_match_obj(match_obj, "value")
+        _value = self.value = get_group_from_match_obj(match_obj, "value")
         self.value_start, self.value_end = get_span_from_match_obj(match_obj, "value")
         self.variable = get_group_from_match_obj(match_obj, "variable")
         self.value_leftquote = get_group_from_match_obj(match_obj, "value_leftquote")
         self.value_rightquote = get_group_from_match_obj(match_obj, "value_rightquote")
         self.clean_url_parameters()
         self.clean_bash_parameters()
         self.sanitize_variable()
+        self.check_value_pos(_value)
+
+    def check_value_pos(self, value: str) -> None:
+        """checks and corrects value_start, value_end in case of self.value was shrink"""
+        if 0 <= self.value_start and 0 <= self.value_end and len(self.value) < len(value):
+            start = value.find(self.value)
+            self.value_start += start
+            self.value_end = self.value_start + len(self.value)
```

Comment:
What about add `sanitize_value()` and move other methods into it?
It may refactor like below. Please consider.

```suggestion
        self.value = get_group_from_match_obj(match_obj, "value")
        self.value_start, self.value_end = get_span_from_match_obj(match_obj, "value")
        self.variable = get_group_from_match_obj(match_obj, "variable")
        self.value_leftquote = get_group_from_match_obj(match_obj, "value_leftquote")
        self.value_rightquote = get_group_from_match_obj(match_obj, "value_rightquote")
        self.sanitize_value()
        self.sanitize_variable()

    def sanitize_value():
        _value = self.value
        self.clean_url_parameters()
        self.clean_bash_parameters()
        self.check_value_pos(_value)

    def check_value_pos(self, value: str) -> None:
        """checks and corrects value_start, value_end in case of self.value was shrink"""
        if 0 <= self.value_start and 0 <= self.value_end and len(self.value) < len(value):
            start = value.find(self.value)
            self.value_start += start
            self.value_end = self.value_start + len(self.value)
```